Update governance with new roles

jacobweinstock opened the pull request:
## Description

<!--- Please describe what this PR is going to change -->
Renamed committers to approvers to follow the pattern established
in Kubernetes community. Added the reviewers role to create a
better latter toward making changes to repositories. Added the
Admin role to help with GitHub administration duties.
This clarifies what inactivity is and how it's measured.


rendered version: https://github.com/jacobweinstock/proposals/blob/0024-community-membership/proposals/0024/GOVERNANCE.md

## Why is this needed

<!--- Link to issue you have raised -->

Fixes: #

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->


## How are existing users impacted? What migration steps/scripts do we need?

<!--- Fixes a bug, unblocks installation, removes a component of the stack etc -->
<!--- Requires a DB migration script, etc. -->


## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade


-----
[View rendered proposals/0024/GOVERNANCE.md](https://github.com/jacobweinstock/proposals/blob/0024-community-membership/proposals/0024/GOVERNANCE.md)
[View rendered proposals/0024/README.md](https://github.com/jacobweinstock/proposals/blob/0024-community-membership/proposals/0024/README.md)

jacobweinstock committed:
Add section on inactivity:

This clarifies what activity is and how its measured.

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>

micahhausler commented on proposals/0024/GOVERNANCE.md:
```diff
@@ -102,8 +102,41 @@ The following apply to the repo or the part of the codebase for which one would
+
+### How inactivity is measured
+
+Inactive members are defined as members of the Tinkerbell Organization
+with **no** contributions across any repos within 8 months. This is
+measured by the CNCF [DevStats project].
```
Do we have any mechanism for this? Do we have a job we can run periodically to diff a list of usernames vs recent contributions? Maybe we don't auto-remove, but even if its a monthly job that just scans for activity and reports inactive contributors that might help make this more actionable. 

Maybe we need a list of folks in each role in a yaml file somewhere (Another case to resurrect `github.com/tinkerbell/org`?)

micahhausler replied:
Opened tinkerbell/org#7

micahhausler commented on proposals/0024/GOVERNANCE.md:
```diff
@@ -102,8 +102,41 @@ The following apply to the repo or the part of the codebase for which one would
+### How inactivity is measured
+
+Inactive members are defined as members of the Tinkerbell Organization
+with **no** contributions across any repos within 8 months. This is
```
We can stick with 8 months, but it won't be easy without tooling to easily tell this. The closest time ranges the [CNCF DevStats Activity page for Tinkerbell](https://tinkerbell.devstats.cncf.io/d/66/developer-activity-counts-by-companies?orgId=1&var-period_name=Last%20year&var-metric=contributions&var-repogroup_name=All&var-country_name=All&var-companies=All) has are either the last year or last quarter.
<img width="508" alt="Screen Shot 2021-10-06 at 9 02 54 AM" src="https://user-images.githubusercontent.com/791000/136207668-e5b4b1fe-b3ac-472a-8819-7ddaba4fff7c.png">

displague replied:
I agree that 8 months is somewhat arbitrary and hard to administer value. I'd be fine with a year but I'm also fine to let this stand as worded.

displague replied:
> [](https://github.com/tinkerbell/proposals/pull/50/files#diff-7f2439d5f7cdc18509ed1616be57259b19699dfd46b3b1675c469ee2289a8ce2)[](https://github.com/)measured by the CNCF [DevStats project].

Perhaps 1 year would be best. Again, this is not a blocker.

micahhausler commented on proposals/0024/GOVERNANCE.md:
```diff
@@ -102,8 +102,41 @@ The following apply to the repo or the part of the codebase for which one would
+with **no** contributions across any repos within 8 months. This is
+measured by the CNCF [DevStats project].
+
+**Note:** Devstats does not take into account non-code contributions. If a
```
[Devstats](https://tinkerbell.devstats.cncf.io/d/66/developer-activity-counts-by-companies?orgId=1&var-period_name=Last%20year&var-metric=contributions&var-repogroup_name=All&var-country_name=All&var-companies=All) does have the following metrics:

- Comments
- Commit comments
- Commits
- GitHub Events
- GitHub pushes
- Issue comments
- Issues
- PRs
- Merged PRs
- Review comments
- Contributions

Should we be more clear that any of the above are included? What about mailing list or community meeting activity? Under this model, is that good enough?

jacobweinstock replied:
yes, more clear is a great idea. Let me rethink this a bit.

displague replied:
What I see is that, before this non-code clause, we are asking for "contributions" without calling out what they are specifically. We can always tighten that up later. I think the current wording is fine for now.

tstromberg commented:
@jacobweinstock - what do you think about switching to a time duration that is easily observable? 

https://github.com/google/pullsheet can do arbitrary values, but it requires someone to run it locally.

jacobweinstock commented:
Hey @tstromberg, I've played around with https://github.com/google/pullsheet. I think I can make the time duration different than what cncf dev stats gives us. Let me play with pullsheet a bit to see what numbers I can get.

jacobweinstock commented:
update:

I've found some existing scripts that will query devstats with arbitrary time ranges.
see these:
- https://www.giters.com/cncf/devstats/issues/309
- https://github.com/cncf/devstatscode/blob/master/devel/api_dev_act_cnt_repos.sh

I've modified it a touch to work with tinkerbell repos. This will allow us to query for 8 months.
see here:
https://gist.github.com/jacobweinstock/f63f2dc9f32fe00a0aab08bcd7798bf5

jacobweinstock commented:
I've created a doc outlining one possible way we can go about bootstrapping the roles described in this proposal. I will also bring this up at the next community meeting.

https://docs.google.com/document/d/1Y4RPFZ0smaSLVmK178Q7fStrLQhdi8Y1BrmRmvZHWkU/edit?usp=sharing

jacobweinstock committed:
Update roles and definitions:

Renamed committers to approvers to follow pattern established
in kubernetes community. Added the reviewers role to create a
better latter toward making changes to repositories. Added the
Admin role to help with GitHub administration duties.

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>

micahhausler commented on proposals/0024/GOVERNANCE.md:
```diff
@@ -102,8 +103,135 @@ The following apply to the repo or the part of the codebase for which one would
+an [elevated set of permissions]. These capabilities should not be used by those
+that are not familiar with the current state of the Tinkerbell project.
+
+Therefore members with an extended period away from the project with no activity
```
Could be worth adding a boomerang clause. Life happens and people might have to step away, but given certain timeframes, we could always add them back

displague replied:
I think the statement provided in the PR works as a boomerang if we change the two occurrences of `will` to `may`.

micahhausler commented:
This has been open for 6 months. Given that there have been no objections, I propose that any other required reviewers approve as-is, and we iterate on boomerang clause/role bootstraping in a successive PR/revision

displague commented:
We'll have to revise https://github.com/orgs/tinkerbell/teams to best fit this structure post-merge.  
Anything else that you are aware of @jacobweinstock, @nshalman?

displague reviewed (approved):
The age of this PR and the hands that it has passed through indicate that this PR broadly carries the form of clarification we need on the contributor model to foster a healthy ecosystem. 

This PR is an incremental improvement and formalization of the proposed model that we have been guided by for the past 8 months.

We will need to onboard processes to truly enact this proposal. I expect that we may need to revise the proposal as we do so.


As a CNCF project, there are some patterns that we could follow found in adjacent projects and templates, such as https://github.com/cncf/project-template (contributor ladder and governance-maintainer). While we may draw further inspiration from these projects, what we have here is well modeled after the philosophies that make these other open-source projects successful.

mergify[bot] committed:
Merge branch 'main' into 0024-community-membership
